backend/application/import_observations/parsers/prometheus_trivy/parser.py

```diff
@@ -0,0 +1,167 @@
+
+        if not prometheus_trivy_base_url.endswith("/"):
+            prometheus_trivy_base_url += "/"
+
+        prometheus_trivy_url = prometheus_trivy_base_url + "api/v1/query?query=" + prometheus_trivy_query
+
+        try:
+            print(prometheus_trivy_base_url)
+            response = requests.get(
+                prometheus_trivy_url, timeout=60, verify=False
+            )
+            response.raise_for_status()
+        except Exception as e:
+            return False, [f"Cannot access Prometheus: {str(e)}"], {}
+
+        return True, [], response.json().get("data").get("result")
+
+    def get_observations(self, data: list[dict]) -> list[Observation]:
+        observations = []
+
+        scanner, version = self.get_about()
+        if version:
+            scanner += " / " + version
+
+        for finding in data:
+            component = finding.get("metric", {}).get("resource", "")
+            image = finding.get("metric", {}).get("image_repository", "")
+            component_name = component + ": " + image
+            component_version = finding.get("metric", {}).get("image_tag", "")
+            vuln_title = finding.get("metric", {}).get("vuln_title", "")
+            vulnerability_id = finding.get("metric", {}).get("vuln_id", "")
+            cvss_v3_base_score = finding.get("metric", {}).get("vuln_score")
+            severity = finding.get("metric", {}).get("severity", Severity.SEVERITY_UNKOWN)
+            description = self.get_description(
+                component_version, vulnerability_id, cvss_v3_base_score, vuln_title, image, component
+            )
+
+            state = ""
+            reference_url = ""
+            cwes = ""
+
+            observation = Observation(
+                title=vulnerability_id,
+                description=description,
+                parser_severity=self.get_severity(severity),
+                parser_status=self.get_status(state),
+                vulnerability_id=vulnerability_id,
+                origin_component_name=component_name,
+                origin_component_version=component_version,
+                cvss3_score=cvss_v3_base_score,
+                cwe=self.get_cwe(cwes),
+                scanner=scanner,
+            )
+
+            evidence = []
+            evidence.append("Vulnerability")
+            evidence.append(json.dumps(finding))
+            observation.unsaved_evidences.append(evidence)
+
+            if reference_url:
+                observation.unsaved_references = [reference_url]
+
+            observations.append(observation)
+
+        return observations
+    
+    def get_description(  # pylint: disable=too-many-branches
+        self,
+        component_version: str,
+        vulnerability_id: str,
+        cvss_v3_base_score: str,
+        vuln_title: str,
+        image: str,
+        component: str,
+    ) -> str:
+        description = ""
+
+        description += f"**Component**: {component}\n\n"
```

Comment:
The fieldes that can be set explicitly in the observation, like `component`, `severity`, `vulnerability_id` and `image` should not be replicated in the description.

Reply:
Is there any other way to display these informations in a issue description?

Reply:
These are attributes in the Observation class (https://github.com/MaibornWolff/SecObserve/blob/dev/backend/application/core/models.py#L386-L503), which you can set.